java/src/main/java/dojo/supermarket/model/ReceiptItem.java

```diff
@@ -0,0 +1,50 @@
+package dojo.supermarket.model;
+
+import java.util.Objects;
+
+public class ReceiptItem {
+
+    private final Product product;
+    private final double price;
+    private final double totalPrice;
+    private final double quantity;
```

Comment:
as above - BigDecimal
```suggestion
    private final BigDecimal price;
    private final BigDecimal totalPrice;
    private final BigDecimal quantity;
```